lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -249,6 +249,65 @@ def get_problem_set_grade_distrib(course_id, problem_set, enrollment):
     return prob_grade_distrib
 
 
+def construct_problem_data(prob_grade_distrib, total_student_count, c_subsection, c_unit, c_problem, component):
```

Comment:
Since this is just a refactoring, can we keep the whitespace as before so it's more cleanly organized?